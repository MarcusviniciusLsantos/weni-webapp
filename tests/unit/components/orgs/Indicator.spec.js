import { shallowMount, createLocalVue } from '@vue/test-utils';
import Indicator from '@/components/orgs/indicator.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('Indicator.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Indicator, {
      localVue,
      i18n,
      propsData: {
        steps: 1,
        names: [],
        current: 0,
      },
      mocks: {
        $t: () => 'some specific text',
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('nameForStep()', () => {
    it('Test valid prop for method', async () => {
      const names = [1, 2, 3];
      await wrapper.setProps({ names });
      const response = wrapper.vm.nameForStep(2);
      expect(response).toStrictEqual(names[1]);
    });
  });
});
