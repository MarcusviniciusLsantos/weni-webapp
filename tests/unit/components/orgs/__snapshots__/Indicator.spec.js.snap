// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator.vue renders a snapshot 1`] = `
<div class="weni-step-indicator">
  <div class="weni-step-indicator__step__wrapper">
    <!---->
    <div class="weni-step-indicator__step"><span class="weni-step-indicator__label">

      </span></div>
  </div>
</div>
`;
